src/main/scripts/src/app/shared/perspectives/icon-picker/icon-picker.component.ts

```diff
@@ -0,0 +1,175 @@
+/*
+ * -----------------------------------------------------------------------\
+ * Lumeer
+ *
+ * Copyright (C) since 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * -----------------------------------------------------------------------/
+ */
+
+import {animate, keyframes, state, style, transition, trigger} from '@angular/animations';
+import {Component, EventEmitter, Input, OnDestroy, OnInit, Output} from '@angular/core';
+import {LocalService} from '../../../core/local.service';
+
+@Component({
+  selector: 'icon-picker',
+  templateUrl: './icon-picker.component.html',
+  styleUrls: ['./icon-picker.component.scss'],
+  animations: [
+    trigger('animateHeight', [
+      state('in', style({height: '*'})),
+      transition('void => *', [
+        animate(100, keyframes([
+          style({height: 0, offset: 0}),
+          style({height: '*', offset: 1})
+        ]))
+      ]),
+      transition('* => void', [
+        animate(100, keyframes([
+          style({height: '*', offset: 0}),
+          style({height: 0, offset: 1})
+        ]))
+      ])
+    ])
+  ]
+})
+export class IconPickerComponent implements OnInit, OnDestroy {
+
+  @Output()
+  private colorChange: EventEmitter<string> = new EventEmitter();
+
+  @Input('color')
+  public activeColor: string;
+
+  public selectedColor: string;
+
+  public previewColor(newColor: string): void {
+    this.activeColor = newColor;
+    newColor ? this.colorChange.emit(newColor) : this.colorChange.emit(this.selectedColor);
+  }
+
+  public selectColor(newColor: string): void {
+    this.selectedColor = newColor;
+    this.colorChange.emit(newColor);
+
+    this.change.emit(`selected ${newColor}`);
+  }
+
+  //=====================================================================
```

Comment:
I could split it into color and icon picker components.. but the @Input(), @Output(), ngOnInit and ngOnDestroy have to be in the icon-picker component either way in order to change icon and color of it's parent.

Also while there is a lot of duplicity at the time, that won't be the case after we add the planned features to it.